### PR TITLE
[Backport] Adding Xi-tracks to lostTracks to 10_6_X

### PR DESCRIPTION
**Pull request by AdrianoDee**
#### PR description:

This is a backport of #33891. The same results apply.

#### PR validation:

All tests run. Given no `bParking` wf is included (AFAIK) in the recommended test, the `1304.181`has been tested (successfully). See results [here](https://adiflori.web.cern.ch/adiflori/lostTracks_pr_106X/).

Would add the `1304.181` to the cmsbuild tests.

### CODE REVIEW

**Comment by slava77** [comment]
@cmsbuild please test workflow 1304.181,136.898

File: PhysicsTools/PatAlgos/plugins/PATLostTracks.cc [hunk 1]
```diff
@@ -194,10 +198,28 @@ void pat::PATLostTracks::produce(edm::StreamID, edm::Event& iEvent, const edm::E
     }
   }
   for (const auto& v0 : *lambdas) {
+    double protonCharge = 0;
     for (size_t dIdx = 0; dIdx < v0.numberOfDaughters(); dIdx++) {
       size_t key = (dynamic_cast<const reco::RecoChargedCandidate*>(v0.daughter(dIdx)))->track().key();
       if (trkStatus[key] == TrkStatus::NOTUSED)
         trkStatus[key] = TrkStatus::VTX;
+      if (xiSelection_)
+        protonCharge += v0.daughter(dIdx)->charge() * v0.daughter(dIdx)->momentum().mag2();
```
Comment (slava77) [inline PhysicsTools/PatAlgos/plugins/PATLostTracks.cc:207]:
```suggestion
      protonCharge += v0.daughter(dIdx)->charge() * v0.daughter(dIdx)->momentum().mag2();
```
this part is different from the master version.
Please either drop here or make a follow-up PR in the master branch.

Comment (AdrianoDee) [reply PhysicsTools/PatAlgos/plugins/PATLostTracks.cc]: Preferred to drop it.

**Comment by slava77** [comment]
@cmsbuild please test

**Comment by slava77** [comment]
+reconstruction

- the backport is correct: changes are the same as in #33891 except for the part of enabling the new selection which is restricted here to `(bParking | run2_miniAOD_devel)`
- tests pass and comparisons show no (relevant) differences

**Comment by slava77** [comment]
@cmsbuild please test workflow 1304.181,136.898

to have a reference bParking test output

**Comment by qliphy** [comment]
+1

**Pull request merged**